docs/reference/filebeat/file-identity.md

```diff
@@ -0,0 +1,189 @@
+
+This can be fixed by using a special marker file on your file system instead of asking the OS for a `device ID`.
+
+Users specify a path by setting `file_identity.inode_marker.path`. This path leads to a file that has some short unique text that remains unchanged forever. The content of this file replaces the `device ID` in the `native` file identity. The rest remains the same as `native`.
+
+### `fingerprint`
+
+The most stable file identity implementation at the moment. This file identity is default since Filebeat version 9.0.
+
+This file identity is using file fingerprints produced by the scanner component of the filestream input. Therefore, in order to activate this file identity, it's required to first activate fingerprint in the scanner by setting `prospector.scanner.fingerprint.enabled: true`.
+
+In the context of the filestream input, fingerprint is a hash string computed from a part of the file content.
+
+![Fingerprint File Identity](/reference/filebeat/img/fingerprint-file-identity.png)
+
+By default, the hashed part of the file is the first 1024 bytes (`offset: 0`, `length: 1024`). Most of the log files have timestamps and other unique segments on each line that make sure that the first 1024 bytes combined together constitute a unique set of characters. This makes sure that a produced fingerprint is a stable unique identifier for each file.
+
+If necessary it's possible to move this hashed byte range farther in the file by adjusting `prospector.scanner.fingerprint.offset`. It's also possible to change the amount of bytes used for hashing by adjusting `prospector.scanner.fingerprint.length` (minimum value is 64 bytes). Keep in mind, the smaller the range, the more likely it is to hit a collision, so the fingerprint (ID) is not unique anymore. If this happens only the first file with a collided ID will be ingested.
+
+This file identity covers most of the use-cases, it's OS-agnostic, supports any file system including network shares and it makes sure to avoid data duplication or data loss.
+
+However, there is a trade off: using this file identity delays ingesting data from files smaller than the required amount of bytes for computing the fingerprint.
+
+For example, Filebeat is running a filestream input with this configuration:
+
+```yaml
+prospector.scanner.fingerprint:
+  enabled: true
+  offset: 32   # e.g. the first 32 bytes are always identical in every file, we skip them
+  length: 1024 # strong guarantee of uniqueness
+file_identity.fingerprint: ~
+```
+
+If a new log file appears, the filestream input would not "see" it until it grows to 1056 bytes (`offset`+`length`).
+
+For some use-cases it might be a requirement to receive data as soon as possible without delays. For some use-cases files would not grow to such sizes at all and would never be ingested. It's important to consider these limitations accordingly.
```

Comment:
```suggestion
For some use cases, it might be a requirement to receive data as soon as possible without delays. For other use cases, files would not grow to such sizes at all and would never be ingested. It's important to consider these limitations accordingly.
```